### PR TITLE
Enable "too many results" stack height validation

### PR DESCRIPTION
**Pull request by chfast**
Requires #355.

### CODE REVIEW

File: test/unittests/validation_stack_test.cpp [hunk 1]
```diff
@@ -160,7 +180,7 @@ TEST(validation_stack, loop_with_result_stack_underflow)
     EXPECT_THROW_MESSAGE(parse(wasm), validation_error, "stack underflow");
 }
 
-TEST(validation_stack, DISABLED_loop_too_many_results)
+TEST(validation_stack, loop_too_many_results)
```
Comment (gumb0) [inline test/unittests/validation_stack_test.cpp:183]: Another test suggestion, not sure if already exists: unreachable insrtuction after `br` that would push extra result, but should be valid, as it's unreachable.

Comment (chfast) [reply test/unittests/validation_stack_test.cpp:183]: This is actually invalid. You cannot have additional items in polymorphic stack.

File: circle.yml [hunk 2]
```diff
@@ -252,8 +252,8 @@ jobs:
       - benchmark:
           min_time: "0.01"
       - spectest:
-          expected_passed: 4903
-          expected_failed: 529
+          expected_passed: 4997
```
Comment (gumb0) [inline circle.yml:255]: wow nice increase

My PRs are getting like 2 new tests passed :confounded:

File: test/unittests/validation_stack_test.cpp [hunk 3]
```diff
@@ -482,6 +502,33 @@ TEST(validation_stack, unreachable_call_indirect)
     parse(wasm);
 }
 
+TEST(validation_stack, unreachable_too_many_results)
+{
+    // TODO: These are actually examples of invalid wasm.
```
Comment (gumb0) [inline test/unittests/validation_stack_test.cpp:507]: Then maybe better to make it disabled tests expecting exception?

File: circle.yml (hunk 2 above)
Comment (chfast) [reply circle.yml:255]: This is part of type validation, so I expect it has a lot of test cases.

**Pull request merged**